apps/meteor/client/providers/AppsProvider/storeQueryFunction.ts

```diff
@@ -0,0 +1,63 @@
+import { type UseQueryResult } from '@tanstack/react-query';
+
+import type { App } from '../../views/marketplace/types';
+
+const sortByName = (apps: App[]): App[] => apps.sort((a, b) => (a.name.toLowerCase() > b.name.toLowerCase() ? 1 : -1));
+
+/**
+ * Aggregates result data from marketplace request and instance installed into their appropriate lists
+ *
+ * Exporting for better testing
+ */
+export function storeQueryFunction(
+	marketplace: UseQueryResult<App[], unknown>,
+	instance: UseQueryResult<App[], unknown>,
+): [App[], App[], App[]] {
+	if (!marketplace.isFetched && !instance.isFetched) {
+		throw new Error('Apps not loaded');
+	}
+
+	const marketplaceApps: App[] = [];
+	const installedApps: App[] = [];
+	const privateApps: App[] = [];
+	const clonedData = [...(instance.data || [])];
+
+	sortByName(marketplace.data || []).forEach((app) => {
+		const appIndex = clonedData.findIndex(({ id }) => id === app.id);
+		const [installedApp] = appIndex > -1 ? clonedData.splice(appIndex, 1) : [];
+
+		const record = {
+			...app,
+			...(installedApp && {
+				private: installedApp.private,
+				installed: true,
+				status: installedApp.status,
+				version: installedApp.version,
+				licenseValidation: installedApp.licenseValidation,
+				migrated: installedApp.migrated,
+			}),
+			bundledIn: app.bundledIn,
+			marketplaceVersion: app.version,
+		};
+
+		if (installedApp) {
+			if (installedApp.private) {
+				privateApps.push(record);
+			} else {
+				installedApps.push(record);
+			}
+		}
+
+		marketplaceApps.push(record);
+	});
+
+	sortByName(clonedData).forEach((app) => {
+		if (app.private) {
+			return privateApps.push(app);
```

Comment:
does Array.prototype.push return anything? it looks confusing this way, I think we could split the return and the push in two different lines, WDYT?

Reply:
It does return the new length of the array after the push. Since it is being returned to the foreach function, I'd say it's pretty ok as foreach will discard it anyway.

Reply:
```suggestion
			privateApps.push(app);
			return;
```